### DISCUSSION TITLE
How to wait until the workflow finishes

### DISCUSSION THREAD

**Discussion by davidfrankoproceedy** [Netflix/conductor#3502] [Q&A]
We are using Netflix Conductor to orchestrate logic in our microservice architecture. Some operations are complex, therefore we are considering using Conductor as an orchestrator for what needs to be done.

The problem we are currently facing is the reactivity of the UI depending on the workflow state. In other words, we would like to start a workflow via rest and receive a response back once it has finished. For example, let's say we have a workflow to send an email, and there are multiple tasks with workers included. We want to start the workflow and send a response back to UI once it has finished. We could implement this by fetching the workflow every x seconds, verifying whether it is in the terminal state.

- Is there some native implementation of this, where we don't have to use pooling?
- Is this a good idea to solve our problem and is Conductor the optimal solution?

Thanks for the help ✌️

**Comment by v1r3n** [comment]
Hi @davidfrankoproceedy we built this in orkes hosted version that does achieve what you need without doing the polling. 
You can give it a try at https://play.orkes.io/ and if you need to discuss further happy to [chat](https://join.slack.com/t/orkes-conductor/shared_invite/zt-xyxqyseb-YZ3hwwAgHJH97bsrYRnSZg).
